google-cloud-pubsub/test/google/cloud/pubsub/topic_test.rb

```diff
@@ -154,6 +154,34 @@
     _(sub).must_be_kind_of Google::Cloud::PubSub::Subscription
   end
 
+  it "creates a subscription with an authenticated push endpoint via push config" do
+    new_sub_name = "new-sub-#{Time.now.to_i}"
+    endpoint = "http://foo.bar/baz"
+
+    push_config = Google::Cloud::PubSub::Subscription::PushConfig.new
```

Comment:
Just noting that this currently `@private` constructor would need to be publicly documented.